### DISCUSSION TITLE
openAMP for socket based

### DISCUSSION THREAD

**Discussion by balrajexploration** [OpenAMP/open-amp#357] [Q&A]
Is it possible to use socket based communication between 2 asymmetrical processors running 2 different os using openAmp?
For example, in SOC has realtime core which runs FreeRtos with realtime applications and application cores which runs linux, so Realtime core applications needs  to send data periodically , then linux core has to receive this data....it would portable if we have socket based interface to applications .. please propose whether Tcp ip on top of rpmsg layer ??? Or any other mechanism ?

**Comment by xiaoxiang781216** [comment]
Yes, you can reference:
https://github.com/apache/incubator-nuttx/tree/master/net/rpmsg
The usage like unix domain socket.

**Reply by balrajexploration** [reply]
it means, there are 3 layers involved  like virt-io+rpmsg+rpmsg_net_socket. What about in freeRTOS side ? shall we have same socket interface

**Reply by balrajexploration** [reply]
But in this case,  which layer does transport functionality like how TCP does ? whats the difference between TCP-IP on virtio

**Reply by xiaoxiang781216** [reply]
> it means, there are 3 layers involved like virt-io+rpmsg+rpmsg_net_socket. What about in freeRTOS side ? shall we have same socket interface

freeRTOS don't have net subsytem, you have to build one by yourself, sorry.

> But in this case, which layer does transport functionality like how TCP does ?

vritio layer.

> whats the difference between TCP-IP on virtio

net/rpmsg provide a socket like interface, so application/service run on different core can talk each other with general socket api instead of rpmsg specific api.

Why do you want to add TCP/IP layer between two core communication? I don't see any benefit here.
You can get more info from: https://cwiki.apache.org/confluence/display/NUTTX/NuttX+And+OpenAMP
